tests/bluetooth/audio/ascs/src/main.c

```diff
@@ -136,6 +149,104 @@ ZTEST_F(ascs_test_suite, test_sink_ase_read_state_idle)
 	zassert_equal(0x00, hdr.ase_state, "unexpected ASE_State 0x%02x", hdr.ase_state);
 }
 
+ZTEST_F(ascs_test_suite, test_cb_register_without_ascs_registered)
+{
+	int err;
+
+	/* Unregister ASCS, as its registered through setup */
+	bt_bap_unicast_server_unregister();
+
+	err = bt_bap_unicast_server_register_cb(&mock_bap_unicast_server_cb);
+
+	zassert_equal(err, -ENOTSUP, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_register_with_null_param)
+{
+	int err;
+
+	/* Unregister ASCS, as its registered through setup */
+	bt_bap_unicast_server_unregister();
+
+	err = bt_bap_unicast_server_register(NULL);
+
+	zassert_equal(err, -EINVAL, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_register_twice)
+{
+	int err;
+	struct bt_bap_unicast_server_register_param param = {
+		CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT,
+		CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT
+	};
+
+	/* Setup already registered once, so calling once here should be sufficient */
+	err = bt_bap_unicast_server_register(&param);
+
+	zassert_equal(err, -EALREADY, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_register_too_many_sinks)
+{
+	int err;
+	struct bt_bap_unicast_server_register_param param = {
+		CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT + 1,
+		CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT
+	};
+
+	/* Unregister ASCS, as its registered through setup */
+	bt_bap_unicast_server_unregister();
+
+	err = bt_bap_unicast_server_register(&param);
+
+	zassert_equal(err, -EINVAL, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_register_too_many_sources)
+{
+	int err;
+	struct bt_bap_unicast_server_register_param param = {
+		CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT,
+		CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT + 1
+	};
+
+	/* Unregister ASCS, as its registered through setup */
+	bt_bap_unicast_server_unregister();
+
+	err = bt_bap_unicast_server_register(&param);
+
+	zassert_equal(err, -EINVAL, "unexpected err response %d", err);
+}
+
+ZTEST_F(ascs_test_suite, test_ascs_unregister_with_ases_in_config_state)
+{
+	const struct test_ase_chrc_value_hdr *hdr;
+	const struct bt_gatt_attr *ase;
+	struct bt_bap_stream *stream = &fixture->stream;
+	struct bt_conn *conn = &fixture->conn;
+	struct bt_gatt_notify_params *notify_params;
+	uint8_t ase_id;
+	int err;
+
+	if (IS_ENABLED(CONFIG_BT_ASCS_ASE_SNK)) {
+		ase = fixture->ase_snk.attr;
+		ase_id = fixture->ase_snk.id;
+	} else {
+		ase = fixture->ase_src.attr;
+		ase_id = fixture->ase_src.id;
+	}
+
+	zexpect_not_null(ase);
+	zexpect_true(ase_id != 0x00);
+
+	test_ase_control_client_config_codec(conn, ase_id, stream);
+
+	err = bt_bap_unicast_server_unregister();
+
+	zassert_equal(err, 0, "unexpected err response %d", err);
```

Comment:
This should not return `0`, right? It should return an error since we have a non-idle ASE

Reply:
It should return 0, as the unregister will initiate a release procedure of all ASEs. The same will (and has been prior to this PR) happen when unregistering the callback.

Reply:
Aha. That behavior should be documented. I'm OK with the behavior, but the current documentation does not tell the user that all ASEs and streams will be released if this is done while in a connection. 

I guess we would also get the `released` callback for each ASE here then? We should test for that as well

Reply:
> I guess we would also get the released callback for each ASE here then? We should test for that as well

Good point!